idpf_specification.md

```diff
@@ -6373,36 +6373,65 @@ plane as <span class="mark">VIRTCHNL2_CAP_RDMA.</span> RDMA negotiation
 
-Since RDMA functionality and the dependent driver relies on the PCIE
-Core LAN driver for PCIE resources such as MSIX interrupt there are a
-few messages that are exchanged over virtchannel with the CP for
-reserving MSIX resources, binding them to queues etc. RDMA driver uses
-the IDPFdriver mailbox and virtchannel for initial configuration
-exchange with the CP.
+IDPF driver will negotiate RDMA Specific capability with the Device Control plane. 
+RDMA is a vendor specific capability as opposed to the rest of the IDPF spec compliant vendor agnostic capabilities. IDPF Spec will cover Interface and driver/device flow details to support a vendor specific RDMA driver along with a generic/common vendor agnostic IDPF driver.
+The Spec will carry the flows and Interfaces between the IDPF driver and IDPF Device Control Plane for RDMA which we refer to as South side interfaces. The North side interfaces between the IDPF driver and the RDMA vendor driver are OS specific and will be covered for Linux as OS in an Appendix section to the Spec.
 
+### South Side Interactions with Device Control Plane
+#### RDMA Capability negotiation
+As the IDPF driver comes up, if it does have the RDMA support code in it, it will announce to the Device Control that it is capable of supporting RDMA capability and driver. If the Device supports the RDMA capability along with IDPF Spec and the Control plane policy allows enabling it for an instance of the PF or VF driver it will respond with the RDMA capability enabled bit in the Capabilities bitmap.
+Note: RDMA transport protocol is agnostic to IDPF driver and the transport in use is a negotiation/learning between the RDMA driver and the Device Control plane. The same RDMA vendor driver is assumed to support all Transports that apply to a given device. 
+As an IDPF driver can come up with multiple logical LAN interfaces corresponding to the multiple vPorts created at Init or later at runtime, the Device Control must mark a vport_flag (a new ENABLE_RDMA flag) for a given vPort so that the driver can then instantiate the corresponding RDMA device Interface instance. If the Vport_flags bitmask does not indicate ENABLE_RDMA for any of the vports even if the device level capability grants RDMA to the driver, no RDMA devices/interfaces will be created by the IDPF driver.
+
+Besides the Device Interface level capability for RDMA, the driver is also enabled by Control plane per vport for enabling RDMA support along side LAN or not. This is indicated as a vport flag set in the response to create_vport by the Device Control plane.
 ```C
-VIRTCHNL2_OP_CONFIG_RDMA_IRQ_MAP
+VIRTCHNL2_VPORT_ENABLE_RDMA             = BIT(4),
+```
 
-VIRTCHNL2_OP_RELEASE_RDMA_IRQ_MAP
+####	MMIO Space Learning
+After the RDMA capability Negotiation IDPF driver must learn about the various MMIO memory regions in the memory BAR that it should not map or it should map so that the RDMA driver can map its own regions in the best possible way depending on usage.
```

Comment:
After the RDMA capability negotiation, IDPF driver must learn about the various MMIO memory regions in the memory BAR that it should map. This will allow RDMA driver to map its own regions in the best possible way depending on the usage.